File: project/src/main/java/com/expedia/bookings/itin/services/TNSApi.java
package com.expedia.bookings.itin.services;

import com.expedia.bookings.itin.data.TNSRegisterDeviceResponse;
import com.expedia.bookings.itin.data.TNSRegisterUserDeviceFlightsRequestBody;
import com.expedia.bookings.itin.data.TNSRegisterUserDeviceRequestBody;

import retrofit2.http.Body;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import rx.Observable;

interface TNSApi {

	@Headers("Content-Type: application/json")
	@POST("/register/user/device")
	Observable<TNSRegisterDeviceResponse> registerUserDevice(
		@Body TNSRegisterUserDeviceRequestBody body
	);

	@POST("/register/user/device/flights")
	Observable<TNSRegisterDeviceResponse> registerUserDeviceFlights(
		@Body TNSRegisterUserDeviceFlightsRequestBody body
	);
}

File: project/src/main/java/com/expedia/bookings/itin/services/TNSApi.kt
package com.expedia.bookings.itin.services

import com.expedia.bookings.itin.data.TNSRegisterDeviceResponse
import com.expedia.bookings.itin.data.TNSRegisterUserDeviceFlightsRequestBody
import com.expedia.bookings.itin.data.TNSRegisterUserDeviceRequestBody
import retrofit2.http.Body
import retrofit2.http.Headers
import retrofit2.http.POST
import rx.Observable

internal interface TNSApi {

    @Headers("Content-Type: application/json")
    @POST("/m/api/register/user")
    fun registerUserDevice(
            @Body body: TNSRegisterUserDeviceRequestBody
    ): Observable<TNSRegisterDeviceResponse>

    @POST("/m/api/register/user/flights")
    fun registerUserDeviceFlights(
            @Body body: TNSRegisterUserDeviceFlightsRequestBody
    ): Observable<TNSRegisterDeviceResponse>
}